src/java/main/org/apache/zookeeper/ClientCnxnSocketNIO.java

```diff
@@ -49,8 +50,9 @@
 
     private SocketAddress remoteSocketAddress;
 
-    ClientCnxnSocketNIO() throws IOException {
-        super();
+    public ClientCnxnSocketNIO(ZKClientConfig clientConfig) throws IOException {
```

Comment:
You're probably making this public for test purposes. Instead of making such methods public, we should be having the tests in the appropriate package so that we don't need to use the access modifiers inappropriately.  I say inappropriately because this method shouldn't really be exposed to applications, it should not be public.